<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="1" ignored="0" total="8" passed="7">
  <reporter-output>
    <line>
      <![CDATA[[INFO] : Assertion Passed : Add New Entry functionality is working fine]]>
    </line>
    <line>
      <![CDATA[[INFO] : Assertion Passed : Not able to add duplicate number]]>
    </line>
    <line>
      <![CDATA[[INFO] : Assertion Passed : Able to remove entry from collection]]>
    </line>
    <line>
      <![CDATA[[INFO] : Assertion Passed : Can handle remove parameter if collection is empty]]>
    </line>
    <line>
      <![CDATA[[INFO] : Assertion Passed : Collection length is correct]]>
    </line>
    <line>
      <![CDATA[[INFO] : Assertion Passed : Function returns position of entry in collection]]>
    </line>
    <line>
      <![CDATA[[INFO] : Assertion Passed : Function returns -1 if entry is not present]]>
    </line>
  </reporter-output>
  <suite name="Default suite" duration-ms="83" started-at="2018-07-09T09:29:55Z" finished-at="2018-07-09T09:29:55Z">
    <groups>
    </groups>
    <test name="Default test" duration-ms="83" started-at="2018-07-09T09:29:55Z" finished-at="2018-07-09T09:29:55Z">
      <class name="testquestion2.Test_MyCollection">
        <test-method status="PASS" signature="Test1_Able_To_Add_New_Entry_In_Collection()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test1_Able_To_Add_New_Entry_In_Collection" duration-ms="15" started-at="2018-07-09T09:29:55Z" description="Check Add functionality" finished-at="2018-07-09T09:29:55Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO] : Assertion Passed : Add New Entry functionality is working fine]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test1_Able_To_Add_New_Entry_In_Collection -->
        <test-method status="PASS" signature="Test2_Duplicate_Entry_In_Collection()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test2_Duplicate_Entry_In_Collection" duration-ms="3" started-at="2018-07-09T09:29:55Z" description="Check Add functionality" finished-at="2018-07-09T09:29:55Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO] : Assertion Passed : Not able to add duplicate number]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test2_Duplicate_Entry_In_Collection -->
        <test-method status="FAIL" signature="Test3_Get_Collection_Functionality()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test3_Get_Collection_Functionality" duration-ms="1" started-at="2018-07-09T09:29:55Z" description="Check toString function" finished-at="2018-07-09T09:29:55Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at java.lang.String.concat(String.java:2027)
at question2.MyCollection.toString(MyCollection.java:145)
at testquestion2.Test_MyCollection.Test3_Get_Collection_Functionality(Test_MyCollection.java:32)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- Test3_Get_Collection_Functionality -->
        <test-method status="PASS" signature="Test5_Remove_Functionality_If_Entry_Already_Present()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test5_Remove_Functionality_If_Entry_Already_Present" duration-ms="2" started-at="2018-07-09T09:29:55Z" description="Check remove functionality" finished-at="2018-07-09T09:29:55Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO] : Assertion Passed : Able to remove entry from collection]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test5_Remove_Functionality_If_Entry_Already_Present -->
        <test-method status="PASS" signature="Test6_Remove_Functionality_If_Collection_Is_Empty()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test6_Remove_Functionality_If_Collection_Is_Empty" duration-ms="1" started-at="2018-07-09T09:29:55Z" description="Check remove functionality" finished-at="2018-07-09T09:29:55Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO] : Assertion Passed : Can handle remove parameter if collection is empty]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test6_Remove_Functionality_If_Collection_Is_Empty -->
        <test-method status="PASS" signature="Test7_Collection_Length_Functionality()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test7_Collection_Length_Functionality" duration-ms="4" started-at="2018-07-09T09:29:55Z" description="Size of Collection" finished-at="2018-07-09T09:29:55Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO] : Assertion Passed : Collection length is correct]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test7_Collection_Length_Functionality -->
        <test-method status="PASS" signature="Test8_Search_Functionality_If_Enrty_Exists()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test8_Search_Functionality_If_Enrty_Exists" duration-ms="5" started-at="2018-07-09T09:29:55Z" description="Search Functionality" finished-at="2018-07-09T09:29:55Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO] : Assertion Passed : Function returns position of entry in collection]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test8_Search_Functionality_If_Enrty_Exists -->
        <test-method status="PASS" signature="Test9_Search_Functionality_If_Enrty_Not_Exists()[pri:0, instance:testquestion2.Test_MyCollection@56f4468b]" name="Test9_Search_Functionality_If_Enrty_Not_Exists" duration-ms="3" started-at="2018-07-09T09:29:55Z" description="Search Functionality" finished-at="2018-07-09T09:29:55Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO] : Assertion Passed : Function returns -1 if entry is not present]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test9_Search_Functionality_If_Enrty_Not_Exists -->
      </class> <!-- testquestion2.Test_MyCollection -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
